### DISCUSSION TITLE
New Anti-Slop Sampling only allows up to 48 banned words/phrases?

### DISCUSSION THREAD

**Discussion by Jonseed** [LostRuins/koboldcpp#1166] [Q&A]
_I'm excited for the new Anti-Slop Sampling!_ The [release page](https://github.com/LostRuins/koboldcpp/releases/tag/v1.76) says: "Up to 48 phrase banning sequences can be used, they are not case sensitive."

Do you know why only up to 48 banned sequences? The default list at [antislop-sampler](https://github.com/sam-paech/antislop-sampler) contains 500 sequences. Their full list contains 50,000+.

Will we be able to soon provide this list as a file instead of a ||$|| delimited string, like the .json lists at antislop-sampler?

**Comment by LostRuins** [comment]
I will increase the limit, but 50k will have an unrealistic overhead. A pruned list will be necessary

**Reply by jabberjabberjabber** [reply]
Will this have side effects for generations not using the anti-slop?

**Reply by Jonseed** [reply]
The 500 word list would probably be sufficient.

**Reply by LostRuins** [reply]
Latest version has greatly increased the limit to 512

**Reply by Jonseed** [reply]
Great! Do we still have to list the sequences in a ||$|| delimited string?

**Reply by LostRuins** [reply]
From Kobold Lite, yes, but you can use the "Add New" button to do it automatically
![image](https://github.com/user-attachments/assets/c26aeea3-dc66-48a5-a7a9-56c232d34a0f)

From the API, no need, it accepts an array in the JSON payload
